app/models/piece.rb

```diff
@@ -124,13 +124,21 @@ def never_moved?
     updated_at == created_at
   end
 
+  def active_player
+    if self.game.move_counter % 2 == 0
+      return self.game.white_user
+    end
+    return self.game.black_user
+  end
+
   def move_to!(x,y)
-    return false unless valid_move?(x,y)
-    capture_piece_at(x,y)
+    return false unless valid_move?(x, y) && self.active_player.id == self.user_id
```

Comment:
This allows movement of the currently active player's piece, but doesn't check to see if the user making the move is the current player.  You will actually need to do that check inside the controller code instead of the model, because models don't conceptually have any knowledge of the "request" that Rails is handling, so they won't have access to current_user and won't be able to validate it.